tools/make_patch.c

```diff
@@ -0,0 +1,597 @@
+	FILE *output = xfopen(output_filename, 'w');
+
+	struct Patches *patches = create_patches();
+
+	// The ROM checksum and Stadium data will always differ
+	append_patch(patches, 0x14e, 2);
+	append_patch(patches, 0x1ffde0, 0xffffff - 0x1ffde0);
+
+	char *buffer = create_buffer();
+	size_t buffer_index = 0;
+	const struct Symbol *current_hook = NULL;
+	int line_pos = 0;
+
+	// Fill in the template
+	for (int c = getc(file); c != EOF; c = getc(file)) {
+		buffer = expand_buffer(buffer, buffer_index);
+
+		switch (c) {
+		case '\r':
+		case '\n':
+			// A newline simply resets the line_pos
+			putc(c, output);
+			line_pos = 0;
+			break;
+
+		case '{':
+			// Check if we've found two of them
+			c = getc(file);
+			if (c != '{') {
+				putc('{', output);
+				line_pos++;
+				ungetc(c, file);
+				break;
+			}
+			// If we have, we store the contents before it ends into buffer
+			buffer_index = 0;
+			for (c = getc(file); c != EOF; c = getc(file)) {
+				if (c == '}') {
+					c = getc(file);
+					if (c == EOF || c == '}') {
+						break;
+					}
+				}
+				buffer[buffer_index++] = c;
+				buffer = expand_buffer(buffer, buffer_index);
+			}
+			buffer[buffer_index] = '\0';
+			interpret_command(buffer, current_hook, symbols, patches, new_rom, orig_rom, output);
+			break;
+
+		case '[':
+			// End immediately if this is not the beginning of the line
+			if (line_pos) {
+				putc(c, output);
+				line_pos++;
+				break;
+			}
+			// Try to read the label
+			putc(c, output);
+			buffer_index = 0;
+			for (c = getc(file); c != EOF; c = getc(file)) {
+				putc(c, output);
+				if (c == ']') {
+					// If we're at the end, we can get the symbol for the label
+					buffer[buffer_index] = '\0';
+					// Translate spaces to underscores
+					for (char *p = buffer; *p != '\0'; p++) {
+						if (*p == ' ') {
+							*p = '_';
+						}
+					}
+					// Look for the symbol starting with .VC_
+					char *searchlabel = xmalloc(strlen(".VC_") + strlen(buffer) + 1);
+					strcpy(searchlabel, ".VC_");
+					strcat(searchlabel, buffer);
+					current_hook = find_symbol(symbols, searchlabel);
+					if (!current_hook) {
+						fprintf(stderr, "Error: Cannot find symbol: %s\n", searchlabel);
+					}
+					free(searchlabel);
+					// Skip until the next newline
+					for (c = getc(file); c != EOF; c = getc(file)) {
+						putc(c, output);
+						if (c == '\n' || c == '\r') {
+							break;
+						}
+					}
+					buffer_index = 0;
+					break;
+				}
+				buffer[buffer_index++] = c;
+			}
+			break;
+
+		default:
+			putc(c, output);
+			line_pos++;
+		}
+	}
+
+	rewind(orig_rom);
+	rewind(new_rom);
+
+	fclose(file);
+	fclose(output);
+	free_buffer(buffer);
+	return patches;
+}
+
+int compare_patch(const void *patch1, const void *patch2) {
+	unsigned int patch1_offset = ((const struct Patch *)patch1)->offset;
+	unsigned int patch2_offset = ((const struct Patch *)patch2)->offset;
+	return patch1_offset > patch2_offset ? 1 : patch1_offset < patch2_offset ? -1 : 0;
```

Comment:
```suggestion
	unsigned int offset1 = ((const struct Patch *)patch1)->offset;
	unsigned int offset2 = ((const struct Patch *)patch2)->offset;
	return offset1 > offset2 ? 1 : offset1 < offset2 ? -1 : 0;
```

Reply:
Resolved in https://github.com/pret/pokecrystal/pull/882/commits/b320e2f9a9ca8768e58082c1240b22c18d9fdfeb